src/main/resources/public/js/getAndAddContracts.js

```diff
@@ -489,19 +489,44 @@ var projectContractsCount = -1;
                     }
                 });
                 if(valid) {
-                    var formData = $(this).serialize();
-                    //check if username exists before submit
-                    usersService.exists(
-                        $("#username").val(),
-                        "github",
-                        function(){
+                    var form = $(this).serialize();
+                    const formData = new FormData(this) 
+                    // check if a contract with the same username and role already exist.
+                    contractsService.getAll(
```

Comment:
@fellahi-ali  Small tip: I think we could do this by simply just query the contracts table instead of calling the server. Table has already a primary key (`rowId` func) registered as `<contributor><role>`.
This pk will ensure quering even if the row is not shown (filtered, is on other page etc..).
```javascript
var exists = $("#contracts").DataTable().row("#aliloDEV").length > 0;
if(exists) reject(...)
```

Reply:
@criske great! that would be more clean and efficient, many thanks :)

Reply:
I also agree. Querying the server for this seems useless :D 
I mean, if the user tries to edit the web page and hack the application, it doesn't really matter what error message we show them. 

So, if they modify the table and then we give them a generic error message, that's their problem.

Reply:
@amihaiemil totally agree and fixed accordingly :)